build.rs

```diff
@@ -121,10 +121,29 @@ fn validate_ident(string: &str) -> StrResult<()> {
 /// Extracts either a single char or parses a U+XXXX escape.
 fn decode_char(text: &str) -> StrResult<char> {
     if let Some(hex) = text.strip_prefix("U+") {
-        u32::from_str_radix(hex, 16)
+        let (hex, name) = match hex.split_once(char::is_whitespace) {
+            Some((hex, name)) => (hex, Some(name.trim_start())),
+            None => (hex, None),
+        };
+
+        let ch = u32::from_str_radix(hex, 16)
             .ok()
-            .and_then(|n| char::try_from(n).ok())
-            .ok_or_else(|| format!("invalid unicode escape {text:?}"))
+            .and_then(|n| char::from_u32(n))
+            .ok_or_else(|| format!("invalid unicode escape {hex:?}"))?;
+
+        #[allow(unused_variables)]
```

Comment:
```suggestion
        #[expect(unused_variables)]
```
We may want to use `expect` here.

Reply:
In this case `allow` is correct because the variable will not be unused if the feature is enabled. This might be a case for `cfg_attr` though—let me try that

Reply:
You're right

Reply:
expect might've also screwed with Typst's MSRV